topostats/processing.py

```diff
@@ -816,7 +837,7 @@ def run_splining(
                         f"[{filename}] : No grains exist for the {direction} direction. Skipping disordered_tracing for {direction}."
                     )
                     splining_grainstats = create_empty_dataframe()
```

Comment:
Perhaps worth being explicit here rather than relying on defaults which might change.

Reply:
Now I've checked the context I don't see the point in creating these empty dataframes here either since they are within `if not ordered_tracing_direction_data:` which immediately after creation will `raises ValueError("No grains exist for the {direction} direction")`.

Could we perhaps create the two empty dataframes when they are first instantiated on lines 808 and 809 and move these two lines to there so instead of...

```python
    splining_grainstats = pd.DataFrame()
    splining_molstats = pd.DataFrame()
```

...we have...

```python
    splining_grainstats = create_empty_dataframe(column_set="grainstats", index_col="grain_number")
    splining_molstats = create_empty_dataframe(column_set="mol_statistics", index_col="molecule_number")
```

:thinking:

Reply:
Tried that, it didn't turn out well because further down there are `merge()` and duplicate columns are introduced.

Reply:
Shall we leave them as is then?

Reply:
Its a bit more work to change the logic and flow of dataframe generation and subsequent merging.

Welcome to tackle it now but I'd be fine with leaving it for future, its not detrimental.

Reply:
Think we'll leave it for the future and just get the fix out asap

Reply:
Could you log it as an issue then please. I know I'll forget unless its written down.